hoptimator-k8s/src/main/java/com/linkedin/hoptimator/k8s/K8sContext.java

```diff
@@ -56,9 +63,18 @@ public SharedInformerFactory informerFactory() {
     return informerFactory;
   }
 
+  public String watchNamespace() {
+    return watchNamespace;
+  }
+
+  public void watchNamespace(String watchNamespace) {
+    this.watchNamespace = watchNamespace;
+  }
+
   public <T extends KubernetesObject, U extends KubernetesListObject> void registerInformer(
       K8sApiEndpoint<T, U> endpoint, Duration resyncPeriod) {
-    informerFactory.sharedIndexInformerFor(generic(endpoint), endpoint.elementType(), resyncPeriod.toMillis());
+    LOG.info("Registering resource={} on namespace={}", endpoint.kind(), namespace);
+    informerFactory.sharedIndexInformerFor(generic(endpoint), endpoint.elementType(), resyncPeriod.toMillis(), watchNamespace);
```

Comment:
Can we just add `watchNamespace` as a parameter to `registerInformer`? This method is only invoke from the operator, which now knows what namespace to watch via the command-line option.